### PR TITLE
Allow more contextual info to be set for font buttons, use correctly in layout widgets

### PR DESCRIPTION
**Pull request by nyalldawson**
Gives more font/symbol buttons access to the message bar and other useful contextual information.

### CODE REVIEW

File: src/gui/qgslabelinggui.cpp [hunk 1]
```diff
@@ -135,8 +135,7 @@ void QgsLabelingGui::updateCalloutWidget( QgsCallout *callout )
 }
 
 QgsLabelingGui::QgsLabelingGui( QgsVectorLayer *layer, QgsMapCanvas *mapCanvas, const QgsPalLayerSettings &layerSettings, QWidget *parent, QgsWkbTypes::GeometryType geomType )
-  : QgsTextFormatWidget( mapCanvas, parent, QgsTextFormatWidget::Labeling )
-  , mLayer( layer )
```
Comment (nirvn) [inline src/gui/qgslabelinggui.cpp]: Don't you need to keep this line?

**Review by nirvn — APPROVED** [review]

File: src/gui/qgslabelinggui.cpp (hunk 1 above)
Comment (nyalldawson) [reply src/gui/qgslabelinggui.cpp]: Ah - thanks, I missed off removing the mLayer member from QgsLabelingGui. It's been moved to the QgsTextFormatWidget base class.

Comment (nyalldawson) [reply src/gui/qgslabelinggui.cpp]: Fixed

**Comment by nirvn** [comment]
Looks good to merge whenever travis turns green.

**Pull request merged**